input/input.go

```diff
@@ -74,6 +74,25 @@ func (i *inputKit) HeightHint() uint32 {
 	return i.heightHint
 }
 
+// BlocksToMaturity returns the relative timelock, as a number of blocks, that
+// must be built on top of the confirmation height before the output can be
+// spent. For non-CSV locked inputs this is always zero.
+func (i *inputKit) BlocksToMaturity() uint32 {
+	return 0
+}
+
+// CraftInputScript returns a valid set of input scripts allowing this output
+// to be spent. The returns input scripts should target the input at location
+// txIndex within the passed transaction. The input scripts generated by this
+// method support spending p2wkh, p2wsh, and also nested p2sh outputs.
+func (i *inputKit) CraftInputScript(signer Signer, txn *wire.MsgTx,
+	hashCache *txscript.TxSigHashes, txinIdx int) (*Script, error) {
+
+	witnessFunc := i.witnessType.GenWitnessFunc(signer, i.SignDesc())
+
+	return witnessFunc(txn, hashCache, txinIdx)
+}
+
```

Comment:
Add commit message with rationale for this? I think the original idea was to allow `Input` implementations to provide their own implementations of these functions. But maybe those don't exist, other than `BaseInput`. Or is there some hiding going on now with this change?

Reply:
This was some leftover refactoring from my early trials. But you are right, it's not very clear why I did this. So I only moved the `BlocksToMaturity` to the `inputKit` to avoid some code duplication. Also described that in the commit message.